### PR TITLE
Podman v4.9.0: Release notes and final cherry-picks

### PR DESCRIPTION
**Pull request by mheon**
Not the actual release bump, having issues with my GPG setup.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by vrothberg — APPROVED** [review]
LGTM

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+## 4.9.0
+### Features
+- The `podman farm` suite of commands for multi-architecture builds is now fully enabled and documented.
+- Add a network recovery service to Podman Machine VMs using the QEMU backend to detect and recover from an inoperable host networking issues experienced by Mac users when ran for long periods of time
+
+### Bugfixes
+- Fixed a bug where the HyperV provider for `podman machine` did not forward the API socket to the host machine.
+- Fixed a bug where improperly formatted annotations passed to `podman kube play` could cause Podman to panic.
+- Fixed a bug where `podman system reset` could fail if non-Podman containers (e.g. containers created by Buildaj) were present.
```
Comment (edsantiago) [inline RELEASE_NOTES.md:11]: Typo, 'Buildah', not worth repushing for just for this

**Review by lsm5 — APPROVED** [review]
/lgtm
/approve
/hold

**Review by edsantiago — COMMENTED** [review]
One typo, otherwise LGTM. Is `farm` actually documented now? If so, we need to remove the "don't check farm man pages" exclusion from `hack/xref-helpmsgs-manpages`

**Comment by mheon** [comment]
I think farm is documented? I cherry-picked a commit with a lot of manpages in, certainly.

**Comment by edsantiago** [comment]
Well, you need to repush. Please `git rm -f docs/source/markdown/podman-farm-build.1.md`.

**Comment by edsantiago** [comment]
The bad commit (that git-added podman-farm-build.1.md`) is fee76bda791a10c54cd4935d46a05f4ef5a54745

**Comment by mheon** [comment]
Repushed. Must have missed whatever commit in main fixed that.

**Comment by edsantiago** [comment]
I would encourage you to cherry-pick #21327 as long as you're repushing.

Context: for a long time `podman farm` was a hidden command, so we couldn't do manpage-vs-help checks. And didn't want to anyway because the command was in flux. Now that it's open, let's reenable the check.

Oops, missed my window! It's okay.

**Comment by mheon** [comment]
Hopefully it passes on main, I'm a little nervous that we haven't been testing this before.

**Comment by edsantiago** [comment]
It passes; I checked before submitting. I've been watching the big podman-farm PRs, but must've missed the one where it became unhidden.

**Comment by edsantiago** [comment]
You can't do it this way. You need to either `git-rm` the bad file before it gets added, or in the commit immediately after. Not at the end.

**Comment by mheon** [comment]
OK, merged it into the docs commit and repushed.

File: docs/source/markdown/options/cache-ttl.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+####> If file is edited, make sure the changes
+####> are applicable to all of those.
+#### **--cache-ttl**
+
+Limit the use of cached images to only consider images with created timestamps less than *duration* ago.
+For example if `--cache-ttl=1h` is specified, Buildah considers intermediate cache images which are created
+under the duration of one hour, and intermediate cache images outside this duration is ignored.
+
+Note: Setting `--cache-ttl=0` manually is equivalent to using `--no-cache` in the
+implementation since this means that the user dones not want to use cache at all.
```
Comment (eriksjolund) [inline docs/source/markdown/options/cache-ttl.md:12]: s/dones/don't

File: docs/source/markdown/options/build-context.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+####> are applicable to all of those.
+#### **--build-context**=*name=value*
+
+Specify an additional build context using its short name and its location.
+Additional build contexts can be referenced in the same manner as we access
+different stages in COPY instruction.
+
+Valid values are:
+
+* Local directory – e.g. --build-context project2=../path/to/project2/src (This option is not available with the remote Podman client. On Podman machine setup (i.e macOS and Winows) path must exists on the machine VM)
```
Comment (eriksjolund) [inline docs/source/markdown/options/build-context.md:13]: s/Winows/Windows

File: docs/source/markdown/options/file.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+####> This option file is used in:
+####>   podman build, farm build
+####> If file is edited, make sure the changes
+####> are applicable to all of those.
+#### **--file**, **-f**=*Containerfile*
+
+Specifies a Containerfile which contains instructions for building the image,
+either a local file or an **http** or **https** URL.  If more than one
+Containerfile is specified, *FROM* instructions are only be accepted from the
```
Comment (eriksjolund) [inline docs/source/markdown/options/file.md:9]: _are only be accepted_
(is that correct English?)

**Comment by mheon** [comment]
@umohnani8 Can you take a look at the farm errors here? They don't look like flakes.

**Comment by edsantiago** [comment]
```
<+201ms> # $ podman farm build --cleanup --local=false --authfile /tmp/podman_bats.O8S6MH/authfile.json --tls-verify=false -t localhost:42469/test-image-2 /tmp/podman_bats.UsVc3f
...
         # Error: build: adding images ["docker://localhost:42469/test-image-2@"] to list: invalid reference format
                                                                      --------^
```
That `@` looks weird

**Comment by umohnani8** [comment]
@mheon I am unable to reproduce the error both locally and by spinning up a ci environment. Can we run the test again to see if it happens again? Maybe something else happened with the push (possibly a network error) and we are just seeing an unrelated error message.

**Comment by edsantiago** [comment]
Pressed rerun. [link to original failure](https://api.cirrus-ci.com/v1/artifact/task/4755280831971328/html/farm-podman-fedora-39-rootless-host-sqlite.log.html)

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,5 +1,18 @@
 # Release Notes
 
+## 4.9.0
+### Features
+- The `podman farm` suite of commands for multi-architecture builds is now fully enabled and documented.
+- Add a network recovery service to Podman Machine VMs using the QEMU backend to detect and recover from an inoperable host networking issues experienced by Mac users when ran for long periods of time
```
Comment (TomSweeneyRedHat) [inline RELEASE_NOTES.md:6]:
```suggestion
- Add a network recovery service to Podman Machine VMs using the QEMU backend to detect and recover from inoperable host networking issues experienced by Mac users when running for long periods of time.
```

File: docs/source/markdown/options/file.md (hunk 4 above)
Comment (TomSweeneyRedHat) [reply docs/source/markdown/options/file.md:9]: no, the word "be" should be removed here.  Nice spot!

File: docs/source/markdown/options/hooks-dir.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+####>   podman build, farm build
+####> If file is edited, make sure the changes
+####> are applicable to all of those.
+#### **--hooks-dir**=*path*
+
+Each *.json file in the path configures a hook for buildah build containers. For more details on the syntax of the JSON files and the semantics of hook injection. Buildah currently support both the 1.0.0 and 0.1.0 hook schemas, although the 0.1.0 schema is deprecated.
+
+This option may be set multiple times; paths from later options have higher precedence.
+
+For the annotation conditions, buildah uses any annotations set in the generated OCI configuration.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/options/hooks-dir.md:11]:
```suggestion
For the annotation conditions, Buildah uses any annotations set in the generated OCI configuration.
```

Comment (mheon) [reply docs/source/markdown/options/hooks-dir.md:11]: This is a backport so we can't really fix it here, need to do it on main

**Comment by mheon** [comment]
Test fails consistently, looks like

**Comment by TomSweeneyRedHat** [comment]
The man page comments could be chased later.  Please hit the release note if you need to respin, otherwise that too can be done later.  Looks like the farm test is still struggling.

**Comment by edsantiago** [comment]
Easily reproduces on a 1mt. @umohnani8 ping me for access. This is not something I can possibly fix.

**Comment by umohnani8** [comment]
@mheon this commit is needed also https://github.com/containers/podman/pull/21004/commits. We fixed a bug on the push endpoint where the output was never being returned. 
Once you cherry pick this, the farm tests should pass!

**Comment by mheon** [comment]
Done, TY @umohnani8

**Comment by mheon** [comment]
All comments on the release itself addressed. We can hit the manpage fixes on main.

**Comment by edsantiago** [comment]
Failure is a known flake. Restarted. Hold still in place so
/lgtm

**Pull request merged**